src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -142,23 +159,80 @@ public String getApiUri() {
             for (GHRepository repo : user.listRepositories()) {
-                add(listener, observer, repo);
+                add(listener, observer, repo, apiUrl);
             }
         }
     }
 
-    private void add(TaskListener listener, SCMSourceObserver observer, GHRepository repo) throws InterruptedException {
+    /**
+     * Get a Github instance
+     */
+    protected GitHub getGithubConnection(String apiUrl, StandardCredentials credentials) throws IOException {
+        return Connector.connect(apiUrl, credentials);
+    }
+
+    /**
+     * Creates an API key for the given username/password with the correct permissions to manage webhooks
+     */
+    @RequirePOST
+    public HttpResponse doCreateApiKey(StaplerRequest req) throws Exception {
+        final String username = req.getParameter("username");
+        final String password = req.getParameter("password");
+
+        try {
+            UsernamePasswordCredentialsImpl creds = BaseGitHubWebHookUtil.get().createApiKey(this.apiUri, username, password);
+            JSONObject response = new JSONObject();
+            response.put("id", creds.getId());
+            return respondJson(200, response.toString());
+        }
+        catch(Exception e) {
+            JSONObject response = new JSONObject();
+            response.put("error", e.getMessage());
+            return respondJson(400, response.toString());
+        }
+    }
+
+    /**
+     * Helper to respond with JSON appropriately
+     */
+    private HttpResponse respondJson(final int status, final String json) {
+        return new HttpResponse() {
+            @Override
+            public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object node) throws IOException, ServletException {
+                rsp.setStatus(status);
+                rsp.setContentType("application/json;charset=UTF-8");
+                rsp.getWriter().println(json);
+            }
+        };
+    }
+
+    /**
+     * Log a message to an appropriate logger
+     */
+    public void logMessage(SCMSourceObserver observer, String message, Object ... params) {
+        logMessage(observer.getListener(), message, params);
+    }
+
+    /**
+     * Log a message to an appropriate logger
+     */
+    public void logMessage(TaskListener listener, String message, Object ... params) {
+        listener.getLogger().format(message, params);
+    }
+
+    private void add(TaskListener listener, SCMSourceObserver observer, GHRepository repo, String apiUrl) throws InterruptedException {
         String name = repo.getName();
         if (!Pattern.compile(pattern).matcher(name).matches()) {
-            listener.getLogger().format("Ignoring %s%n", name);
+            logMessage(listener, "Ignoring %s%n", name);
             return;
         }
-        listener.getLogger().format("Proposing %s%n", name);
+        logMessage(listener, "Proposing %s%n", name);
```

Comment:
:ant: because changes the criteria, here an [example](https://github.com/jenkinsci/github-branch-source-plugin/blob/master/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java#L107).